metrics/storage/blogbench.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Description of the test:
+# This test runs the 'blogbench', and extracts the 'scores' for reads
+# and writes
+# Note - the scores are *not* normalised for the number of iterations run,
+# they are total scores for all iterations (this is the blogbench default output)
+
+set -e
+
+# General env
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/../lib/common.bash"
+# Check tools/commands dependencies
+TEST_ARGS="$@"
+
+TEST_NAME="blogbench"
+IMAGE="local-blogbench"
+DOCKERFILE="${SCRIPT_PATH}/blogbench_dockerfile/Dockerfile"
+
+# Number of iterations for blogbench to run - note, results are not
+# scaled to iterations - more iterations results in bigger results
+ITERATIONS="${ITERATIONS:-30}"
+
+# Directory to run the test on
+# This is run inside of the container
+TESTDIR="${TESTDIR:-/tmp}"
+CMD="blogbench -i ${ITERATIONS} -d ${TESTDIR}"
+
+function main() {
+	cmds=("awk" "docker")
+
+	init_env
+	check_cmds "${cmds[@]}"
+	check_dockerfiles_images "$IMAGE" "$DOCKERFILE"
+
+	metrics_json_init
+
+	local output=$(docker run --rm --runtime=$RUNTIME $IMAGE $CMD)
+
+	# Save configuration
+	metrics_json_start_array
+
+	local frequency=$(echo "$output" | grep "Frequency" | cut -d "=" -f2)
```

Comment:
Looks like this line (and some others maybe) don't generate valid JSON (it generates '10 secs', which is not a number, so will need double-quoting for instance).
Other errors I see are the final members of comma separated lists having commas after them. You might have to do a little scripty fun like https://github.com/kata-containers/tests/blob/master/metrics/lib/json.bash#L90-L96 to handle the no-comma final case etc.

My `vim` hightlights these bits of invalid JSON for me, but I believe you can use `jq . blogbench.json` on the results file to pick out invalid JSON syntax.